lib/Runtime/Library/TypedArray.cpp

```diff
@@ -144,10 +144,13 @@ namespace Js
                 if (JavascriptOperators::IsObject(firstArgument))
                 {
                     Var iterator = JavascriptOperators::GetProperty(RecyclableObject::FromVar(firstArgument), PropertyIds::_symbolIterator, scriptContext);
+                    Var arrayIteratorPrototypeNext = nullptr;
 
                     if (!JavascriptOperators::IsUndefinedObject(iterator) &&
                         (iterator != scriptContext->GetLibrary()->GetArrayPrototypeValuesFunction() ||
-                        !JavascriptArray::Is(firstArgument)))
+                            !JavascriptArray::Is(firstArgument) ||
+                            !RuntimeFunction::Is(arrayIteratorPrototypeNext = JavascriptOperators::GetProperty(scriptContext->GetLibrary()->GetArrayIteratorPrototype(), PropertyIds::next, scriptContext)) ||
+                            &JavascriptArrayIterator::EntryInfo::Next != RuntimeFunction::FromVar(arrayIteratorPrototypeNext)->GetFunctionInfo()))
```

Comment:
I'd rather we compare the `JavascriptFunction` objects instead, getting the array iterator next object from the library for the comparison.